Remove ztunnel hairpinning from docs (#14995)
istio/istio.io: content/en/docs/ambient/usage/ztunnel/index.md

caption="Ztunnel datapath via an interim waypoint"
>}}

### Note on HBONE {#hbonesection}

> HBONE (HTTP Based Overlay Network Encapsulation) is an Istio-specific term. It refers to the use of standard HTTP tunneling via the [HTTP CONNECT](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/CONNECT) method to transparently tunnel application packets/ byte streams. In its current implementation within Istio, it transports TCP packets by tunneling these transparently using the HTTP CONNECT method, uses [HTTP/2](https://httpwg.org/specs/rfc7540.html), with encryption and mutual authentication provided by [mutual TLS](https://www.cloudflare.com/learning/access-management/what-is-mutual-tls/) and the HBONE tunnel itself runs on TCP port 15008. The overall HBONE packet format from IP layer onwards is depicted in the following figure.

Reviewer: HBONE (HTTP Based Overlay Network Encapsulation) is an Istio-specific term. It refers to the use of [mutual TLS](https://www.cloudflare.com/learning/access-management/what-is-mutual-tls/) and HTTP tunnels via the [HTTP CONNECT](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/CONNECT) method to transparently carry TCP byte streams. In its current implementation within Istio, this is done with [HTTP/2](https://httpwg.org/specs/rfc7540.html) to enable efficient multiplexing and TLS connection reuse. HBONE is served on the reserved TCP port 15008. The overall HBONE packet format from IP layer onwards is depicted in the following figure.

Reviewer 2: cc @bleggett who is looking at hbone doc.

Author: Updated, PTAL @louiscryan @bleggett

Reviewer: Thanks but I think we can now just update to refer to https://github.com/istio/istio.io/pull/15004

Reviewer 3: (my bad I missed this) - yeah just leave this as is and the other PR will remove this section entirely

### East-West non-mesh pod to ambient mesh pod (and use of PeerAuthentication resource) {#ewnonmesh}

> In the example below, the same `httpbin` service which has already been set up in the prior examples is accessed via client `sleep` pods that are running in a separate namespace that is not part of the Istio mesh. This example shows that East-west traffic between ambient mesh pods and non mesh pods is seamlessly supported. Since the non-mesh pods initiate traffic directly to the backend pods without going through HBONE or ztunnel. Note that in this case, Istio L4 policy can control whether traffic is denied or allowed to bypass the waypoint.

Author: Does this read better @ilrudie @linsun ?

Reviewer 2: answered a bit below.

### East-West non-mesh pod to ambient mesh pod (and use of PeerAuthentication resource) {#ewnonmesh}

> In the example below, the same `httpbin` service which has already been set up in the prior examples is accessed via client `sleep` pods that are running in a separate namespace that is not part of the Istio mesh. This example shows that East-west traffic between ambient mesh pods and non mesh pods is seamlessly supported. Note that in this case, Istio L4 policy can control whether traffic is denied or allowed from the source.

Reviewer 2: Hi @MorrisLaw LGTM.  You may want to address @louiscryan 's comments - https://github.com/istio/istio.io/pull/14995#discussion_r1585317353

> In the example below, the same `httpbin` service which has already been set up in the prior examples is accessed via client `sleep` pods that are running in a separate namespace that is not part of the mesh. This example shows that east-west traffic between ambient mesh pods and non mesh pods is seamlessly supported. Since the non-mesh pods initiate traffic directly to the backend pods without going through ztunnel. Note that in this case, Istio L4 policy can control whether traffic is denied or allowed from the source.

Reviewer 2: ```suggestion
In the example below, the same `httpbin` service which has already been set up in the prior examples is accessed via client `sleep` pods that are running in a separate namespace that is not part of the mesh. This example shows that east-west traffic between ambient mesh pods and non mesh pods is seamlessly supported. The non-mesh pods initiate traffic directly to the destination pods without going through the source ztunnel, while the destination ztunnel enforces any L4 policy to control whether traffic should be allowed or denied.
```

Reviewer 2: How about this?

Author: I agree, clarifying `destination ztunnel` and `source ztunnel` is more clear. Will update.

> In the example below, the same `httpbin` service which has already been set up in the prior examples is accessed via client `sleep` pods that are running in a separate namespace that is not part of the Istio mesh. This example shows that East-west traffic between ambient mesh pods and non mesh pods is seamlessly supported. Since the non-mesh pods initiate traffic directly to the backend pods without going through HBONE or ztunnel. Note that in this case, Istio L4 policy can control whether traffic is denied or allowed to bypass the waypoint.

Reviewer 4: Yeah, seems clearer to me. I'd remove the preceding sentence at this point too though.
```suggestion
In the example below, the same `httpbin` service which has already been set up in the prior examples is accessed via client `sleep` pods that are running in a separate namespace that is not part of the Istio mesh. This example shows that East-west traffic between ambient mesh pods and non mesh pods is seamlessly supported. Note that in this case, Istio L4 policy can control whether traffic is denied or allowed to bypass the waypoint.
```

Reviewer 2: Hi @MorrisLaw @ilrudie I think I'm confused with `Istio L4 policy can control whether traffic is denied or allowed to bypass the waypoint.`  My mental model is this is controlled by source ztunnel based on the source (out of mesh or ingress or sidecar or ztunnel captured pod).  You have L4 policy which is enforced at destination ztunnel thus it is confusing to me that it controls bypass waypoint. I think it should be:

Note that in this case, Istio L4 policy can control whether traffic is denied or allowed from the source (which is sleep here I think?).

Author: Is this policy part, specifically, documented somewhere @linsun ?

>  Istio L4 policy can control whether traffic is denied or allowed from the source

I'm getting a bit confused myself now. I think a reference might be helpful.

Author: That makes sense, thank you for clarifying!

Author: Updated, PTAL @linsun

Reviewer 4: Source ztunnel does control if traffic is sent to a waypoint or not but it's not via an authz policy. The destination ztunnel can enforce things like "this wl can only be reached from my waypoint's identity" or conversely it can be permissive of traffic without an identity or it may allow an identity of a non-waypoint wl to reach the destination.

I was describing that as allowing the waypoint to be "bypassed". I can see how that's a confusing way to think about/describe what's happening though.

> In the example below, the same `httpbin` service which has already been set up in the prior examples is accessed via client `sleep` pods that are running in a separate namespace that is not part of the Istio mesh. This example shows that East-west traffic between ambient mesh pods and non mesh pods is seamlessly supported. Since the non-mesh pods initiate traffic directly to the backend pods without going through HBONE or ztunnel. Note that in this case, you have control over whether or not the waypoint gets bypassed too depending on how you configure policy.

Reviewer 2: I don't think user can control whether waypoint gets bypassed.  The traffic won't be sent to waypoint if it is from out of mesh.

Reviewer 4: +1, I think it's just a wording/clarity issue though. You can control if bypass is allowed or denied using istio L4 policy but the way it is worded seems like it could mean that you can define a policy that makes the traffic hairpin to the waypoint which we do not offer.

Reviewer: nits:

- no need to say 'Istio' 
- s/East-west/east-west
- s/through HBONE or ztunnel/through ztunnel  (we can elaborate on HBONE elsewhere)
